src/login.js

```diff
@@ -126,53 +130,69 @@ async function askAndFillVerificationCodeIfFieldExists(page) {
     }
 }
 
-async function interceptSamlPayloadForAWS(interceptedRequest, page, browser) {
+async function interceptSamlPayloadForAWS(interceptedRequest, page, browser, preselectedRoleArn) {
     if (interceptedRequest._url === 'https://signin.aws.amazon.com/saml') {
         samlPayloadIntercepted = true;
 
         const samlResponse = saml.getSamlResponse(interceptedRequest);
         const samlResponseBase64 = saml.getBase64SamlResponse(samlResponse);
         const samlResponseDecoded = saml.getDecodedSamlResponse(samlResponseBase64);
         const rolesAndPrincipals = saml.getRolesAndPrincipalsFromSamlResponse(samlResponseDecoded);
-        const selectedRoleAndPrincipal = await letUserChooseAwsRole(rolesAndPrincipals);
+
+        const selectedRoleAndPrincipal = await chooseAwsRole(rolesAndPrincipals, preselectedRoleArn);
+
+        if (selectedRoleAndPrincipal === undefined) {
+            throw new Error('Invalid role ARN provided, could not log in.');
+        }
 
         await AwsCliUtil.configureWithSamlAssertion(selectedRoleAndPrincipal.role, selectedRoleAndPrincipal.principal, samlResponseBase64, configure.getAssumeDuration());
         console.log('Successfully logged in, use the profile `keyhub`. (export AWS_PROFILE=keyhub / set AWS_PROFILE=keyhub)');
 
         await interceptedRequest.abort();
         await exit(browser, page);
+
     } else {
         await interceptedRequest.continue();
     }
 }
 
-async function letUserChooseAwsRole(rolesAndPrincipals) {
-    const answer = await inquirer.prompt([
+async function chooseAwsRole(rolesAndPrincipals, preselectedRoleArn) {
```

Comment:
If the role is specified with the --role-arn property there is nothing to choose. The function should be named something like getAwsRole, determineAwsRole e.g.